### PR TITLE
No need to have a "while (TRUE)" statement if we can check at the end of the loop or at the beginning.

### PR DESCRIPTION
**Pull request by ghost**
Whenever there must be a loop that goes indefinitely, there are either return statements or break statements in the code.

However, in this codebase, there are "infinite loops" that appear as such until the next line is checked, which is an if statement saying to break if a condition is true.

This unneeded complexity can be resolved by inverting the condition and making that the while-loop condition.

### CODE REVIEW

File: src/mono/mono/mini/unwind.c [hunk 1]
```diff
@@ -258,7 +258,7 @@ decode_uleb128 (guint8 *buf, guint8 **endbuf)
 	guint32 res = 0;
 	int shift = 0;
 
-	while (TRUE) {
+	for (;;) {
```
Comment (lambdageek) [inline src/mono/mono/mini/unwind.c]: just a personal opinion, but `while (TRUE)` is clearer than `for (;;)`.  I don't think this change is useful.

File: src/mono/mono/mini/unwind.c [hunk 2]
```diff
@@ -280,7 +280,7 @@ decode_sleb128 (guint8 *buf, guint8 **endbuf)
 	gint32 res = 0;
 	int shift = 0;
 
-	while (TRUE) {
+	for (;;) {
```
Comment (lambdageek) [inline src/mono/mono/mini/unwind.c]: same as the one above - `while (TRUE)` feels clearer to me.

File: src/mono/mono/mini/unwind.c [hunk 3]
```diff
@@ -404,8 +404,7 @@ mono_unwind_ops_encode_full (GSList *unwind_ops, guint32 *out_len, gboolean enab
 	p = buf;
 
 	loc = 0;
-	l = unwind_ops;
-	for (; l; l = l->next) {
+	for (l = unwind_ops; l; l = l->next) {
```
Comment (lambdageek) [inline src/mono/mono/mini/unwind.c]: could also move the declaration of `GSList *l` to this for loop
```suggestion
	for (GSlist *l = unwind_ops; l; l = l->next) {
```

**Review by lambdageek — APPROVED** [review]
This is ok, but @pi1024e you're beginning to stray into essentially stylistic differences.  Things like `while (TRUE)` vs `for (;;)` and `if (cond) stmt1; else stmt2;` vs `if (cond) stmt1; stmt2;` are semantically identical and it really comes down to coding style.

Generally, for Mono we are weary of stylistic changes because they obscure `git blame` which makes it more difficult to discover why certain code is the way it is.

IMO there are two good changes in this PR (but again none of them are changing the semantics, so it's a style call on my part) :
1. The change in `mono_unwind_ops_encode_full` to declare and init `l` in the `for`-loop
2. The change in `mono_unwind_decode_llvm_mono_fde` from `while (TRUE) { if (...) break; ...}` to `while (...) {...}`.

**Review by CoffeeFlux — CHANGES_REQUESTED** [review]
The rest of our code uses `while (TRUE)`, so unless there's a compelling reason I don't think we should be changing it to `for (;;)`.

File: src/mono/mono/mini/unwind.c [hunk 4]
```diff
@@ -1167,8 +1166,7 @@ mono_unwind_decode_fde (guint8 *fde, guint32 *out_len, guint32 *code_len, MonoJi
 	while (p < cie + cie_len + 4) {
 		if (*p == DW_CFA_nop)
```
Comment (Clockwork-Muse) [inline src/mono/mono/mini/unwind.c]: Should this condition be inverted and hoisted into the `while`, like what was done at line 1259?

File: src/mono/mono/mini/unwind.c (hunk 1 above)
Comment (pi1024e) [reply src/mono/mono/mini/unwind.c]: for (;;) is a well-known idiom, and having TRUE in all caps can cause issues with some compilers.

File: src/mono/mono/mini/unwind.c (hunk 2 above)
Comment (pi1024e) [reply src/mono/mono/mini/unwind.c]: As mentioned above.

**Comment by pi1024e** [comment]
@CoffeeFlux I changed them back to while (TRUE). Is this appropriate now?

File: src/mono/mono/mini/unwind.c (hunk 4 above)
Comment (pi1024e) [reply src/mono/mono/mini/unwind.c]: No

**Review by CoffeeFlux — APPROVED** [review]
I'm still iffy in general on purely stylistic PRs like this since I don't think it's usually worth the history damage, but this seems fine assuming it passes CI.

**Pull request merged**